src/util/sock.cpp

```diff
@@ -117,6 +117,34 @@ int Sock::GetSockName(sockaddr* name, socklen_t* name_len) const
     return getsockname(m_socket, name, name_len);
 }
 
+bool Sock::SetNonBlocking() const
+{
+#ifdef WIN32
+    u_long on{1};
+    if (ioctlsocket(m_socket, FIONBIO, &on) == SOCKET_ERROR) {
+        return false;
+    }
+#else
+    const int flags{fcntl(m_socket, F_GETFL, 0)};
+    if (flags == SOCKET_ERROR) {
```

Comment:
We were previously not checking for an error on this `fcntl` call. Is that really necessary or would the next `fcntl` call fail when being passed `SOCKET_ERROR | O_NONBLOCK`?

(Not against adding the error check, just wanted to note that i noticed the addition)

Reply:
I do not know if the next `fcntl()` call would fail. The doc is a bit unclear what happens when passed unknown/invalid flags to `fcntl()`. Also, given that `SOCKET_ERROR` is `-1` and `O_NONBLOCK` is `0x0004`, then what is `-1 | 0x0004`? It is `-1` actually, but is definitely something we don't want to do.